.github/actions/release-pr/action.yml

```diff
@@ -0,0 +1,249 @@
+# Workflow to create a new release PR, with one of the following two scenarios:
+#
+# - Major release
+#   - Pushes a new `release/<tag-prefix>-v<version>` branch based on latest `major.minor` version, e.g. `release/aptos-v1.0`
+#   - Creates a new `release-pr` branch from the release, then bumps the version with the `version` input
+#   - Opens a release PR from `release-pr` to `release/<tag-prefix>-v<version>`
+# - Minor release
+#   - Pushes a new `release/<tag-prefix>-v<version>` branch based on the latest compatible major release
+#   - Creates a new `release-pr` branch from the release, then bumps the version with the `version` input
+#   - Opens a release PR from `release-pr` to `release/<tag-prefix>-v<version>`
+# - Patch release
+#   - Pushes a new `patch/<tag-prefix>-v<version>` branch based on `release/<tag-prefix>-v<version>`, then bumps the verision with the `version` input
+#   - Errors if the `release/<tag-prefix>-v<version>` branch doesn't exist
+#   - Opens a release PR from `patch/<tag-prefix>-v<version>` to `release/<tag-prefix>-v<version>`
+#
+# When the PR is merged, the caller can then trigger a release from `ci-workflows/actions/tag-release`
+# The PR branch can then be safely deleted, while the release branch should have a branch protection rule for historical preservation
+#
+# Example caller workflow: TODO: Link to Sphinx, ZKLC
+# TODO: Add unique ID to PR branch name, e.g. release-pr-<version>-<tag-prefix> in order to allow multiple release PRs at once
+name: Create release PR
+
+description: Bump crate versions and open release PR
+
+inputs:
+  # Optional working directory relative to `${{ github.workspace }}`, e.g. "aptos/" in "zk-light-clients/"
+  path:
+    description: 'Relative path under the GitHub workspace'
+    required: false
+  tag-prefix:
+    description: 'Optional tag prefix before version number'
+    required: false
+  more-crates:
+    description: 'Additional non-workspace crate paths to release'
+    required: false
+  # Caller workflows should specify a `choice` input or otherwise ensure the only options are `major`, `minor`, `patch`
+  release-type:
+    description: 'Semver release type: `major`, `minor`, or `patch`'
+    required: true
+    default: 'major'
+  # The release branch will be persistent across patches, so any input `patch` version must correspond to an existing branch
+  version:
+    description: 'Semver version corresponding to `release-type`, e.g. `1.0.0` for `major`'
+    required: true
+  token:
+    description: 'Git token'
+    required: true
```

Comment:
Does this have to be required or could we default on this repo token? Either a secret or the `GITHUB_TOKEN`?

Reply:
We can't use the default `GITHUB_TOKEN` due to the permissions limitation described in https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#token. If we default to a secret, we'd have to give it a name (e.g. `secrets.REPO_TOKEN`), but that would be confusing to the user because there's no standard secret name for PATs.